kolibri/core/auth/serializers.py

```diff
@@ -84,6 +84,21 @@ class Meta:
             "preset",
         )
 
+    def validate(self, attrs):
```

Comment:
Because of the second test case I suggested below, this validation will pass in cases where it should not.

I think it will be more effective (and DRYer) to not use the DRF Serializer validation here, but rather to override the serializer save method, and catch the `IncompatibleDeviceSetting` error that I have suggested should be raised by the model save method.

This would look something like:
```
def save(self, **kwargs):
    try:
        return super(FacilityDatasetSerializer, self).save(**kwargs)
    except IncompatibleDeviceError:
        raise serializers.ValidationError(....
```

Reply:
Yup. Catching the error in `save` makes it simpler. Thank you for the help.